pkg/test/collector_intermediate_test.go

```diff
@@ -33,24 +34,20 @@ func TestCollectorToIntermediate(t *testing.T) {
 	cp, _ := collector.InitCollectingProcess(address, 1024, 0)
 	ap, _ := intermediate.InitAggregationProcess(cp.GetMsgChan(), 2, fields)
 
+	waitForCollectorReady(t, address)
 	go func() {
-		time.Sleep(time.Second)
 		conn, err := net.DialUDP("udp", nil, address)
 		if err != nil {
 			t.Errorf("UDP Collecting Process does not start correctly.")
 		}
 		defer conn.Close()
 		conn.Write(templatePacket)
-		time.Sleep(time.Second)
 		conn.Write(dataPacket1)
-		time.Sleep(time.Second)
 		conn.Write(dataPacket2)
 	}()
+	go ap.Start()
 	go func() {
-		ap.Start()
-	}()
-	go func() {
-		time.Sleep(4 * time.Second)
+		time.Sleep(time.Second)
```

Comment:
is there any indicator from aggregation process that messages and records are ready? Please use that instead of sleep.

In addition you should start cp.Start in go routine wait in the test main to stop collector process and aggregation process.

Reply:
Updated wait with checking whether flowKeyRecordMap gets data. Thanks!